pkg/kapp/resourcesmisc/custom_waiting_resource.go

```diff
@@ -95,6 +97,14 @@ func (s CustomWaitingResource) IsDoneApplying() DoneApplyState {
 					hasConditionWaitingForGeneration = true
 					continue
 				}
+
+				if condMatcher.SupportsUnblockingChanges {
```

Comment:
we should probably do this section after the below if success section. if it's successful, we just want to get out as quick as possible.

Reply:
Done